python/lsst/sconsUtils/scripts.py

```diff
@@ -250,8 +251,8 @@ def python(swigNameList=None, libs="main python", swigSrc=None):
     ##
     @staticmethod
     def doc(config="doxygen.conf.in", projectName=None, projectNumber=None, **kw):
-        if not state.env.ProductDir("doxygen"):
-            state.log.warn("Doxygen is not setup; skipping documentation build.")
+        if not find_executable("doxygen"):
+            state.log.warn("doxygen executable not found; skipping documentation build.")
```

Comment:
I think it'd be more idiomatic to do this check in an `SCons.Configure` context, but I'm not sure if there's any real penalty for doing it this way, especially considering that we just always rebuild the docs instead of limiting that to when we think they've changed, and that's usually the sort of thing that'd be broken by doing configure checks in the wrong way.